### PR TITLE
feat(writers): Support @ui5/cli v3 in the generated projects

### PR DESCRIPTION
**Pull request by docirl**
Issue: #904 

- use @ui5/cli ^3.0.0 in package.json
- use version 2.6 in ui5*.yaml template 
- update odata-service-writer to not add `@sap/ux-ui5-tooling` in ui5 dependancies when `@ui5/cli` v3 is used, add tests
- reverts https://github.com/SAP/open-ux-tools/pull/903
- reverts https://github.com/SAP/open-ux-tools/pull/928
- update tests/snapshots

### CODE REVIEW

**Review by devinea — DISMISSED** [review]
Thanks @docirl 
changesets, code, templates, snapshots and lock file updates look good.

File: packages/odata-service-writer/src/updates.ts [hunk 1]
```diff
@@ -43,13 +44,15 @@ export function updateManifest(basePath: string, service: OdataService, fs: Edit
 export function updatePackageJson(path: string, fs: Editor, addMockServer: boolean) {
     const packageJson = JSON.parse(fs.read(path));
     packageJson.devDependencies = packageJson.devDependencies ?? {};
-    packageJson.ui5 = packageJson.ui5 ?? {};
-    packageJson.ui5.dependencies = packageJson.ui5.dependencies ?? [];
+    if (!hasUI5CliV3(packageJson.devDependencies)) {
```
Comment (IainSAP) [inline packages/odata-service-writer/src/updates.ts:47]: Im not sure about the logic here now, its a little difficult to follow as the presence of `pacakgeJson.ui5` is being used as a condition throughout but thats not really the thing thats relevant. I think this function should be refactored to be clearer and remove the conditions `packageJson.ui5` and replace with a more explicit condition.

```suggestion
   const ui5CliVersion = getVersion('@ui5/cli');
   ...
   // Add stuff common to all versions of @ui5/cli
   if (ui5CliVersion < '3') {
       // Add stuff for `ui5` section only
   } 
```

File: packages/odata-service-writer/test/unit/updates.test.ts [hunk 2]
```diff
@@ -41,4 +42,33 @@ describe('updates', () => {
...
+        test('Add @sap/ux-ui5-tooling dependency to ui5 if @ui5/cli version is less than 3.0.0', () => {
+            testPackageJson.devDependencies['@ui5/cli'] = '^2.14.1';
+            const path = join('./test1', packageJsonFile);
+            fs.writeJSON(path, testPackageJson);
+            updatePackageJson(path, fs, false);
+            const packageJson = fs.readJSON('./test1/package.json') as Package;
+            expect(packageJson.ui5?.dependencies).toEqual(['@sap/ux-ui5-tooling']);
+        });
+
+        test('Add @sap/ux-ui5-tooling dependency to ui5 if @ui5/cli version is less than 3.0.0', () => {
```
Comment (IainSAP) [inline packages/odata-service-writer/test/unit/updates.test.ts:65]: Wrong test name?

**Review by IainSAP — COMMENTED** [review]
A few comments...

File: packages/odata-service-writer/src/updates.ts [hunk 3]
```diff
@@ -43,13 +44,15 @@ export function updateManifest(basePath: string, service: OdataService, fs: Edit
...
+        packageJson.ui5 = packageJson.ui5 ?? {};
+        packageJson.ui5.dependencies = packageJson.ui5.dependencies ?? [];
```
Comment (IainSAP) [inline packages/odata-service-writer/src/updates.ts:49]: Should we remove these sections if they are present and the version is 3+?

Comment (docirl) [reply packages/odata-service-writer/src/updates.ts:49]: Similar as discussion from here https://github.com/SAP/open-ux-tools/pull/935/files/69fd7b5348091cc5d60368c56709261a2f960a7c#r1122681360 so I went with the same approach here.

File: packages/odata-service-writer/test/unit/updates.test.ts (hunk 2 above)
Comment (docirl) [reply packages/odata-service-writer/test/unit/updates.test.ts:65]: Thanks, I'll fix that

File: packages/odata-service-writer/src/updates.ts (hunk 1 above)
Comment (docirl) [reply packages/odata-service-writer/src/updates.ts:47]: I'll refactor this to add a block to make updates only when ui5/cli is lower than 3. Afaik the ui5 section is only for cli but not certain.

**Review by IainSAP — APPROVED** [review]
Thx @docirl 

- Review updates look good

**Review by devinea — APPROVED** [review]
Thanks @docirl

**Comment by johannes-kolbe** [comment]
Thanks @docirl, changes look good to me.

**Comment by johannes-kolbe** [comment]
Thanks @docirl, changes look good to me.

**Pull request merged**